builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -178,12 +184,14 @@ func resourceAwsElasticBeanstalkEnvironmentUpdate(d *schema.ResourceData, meta i
 func resourceAwsElasticBeanstalkEnvironmentDescriptionUpdate(conn *elasticbeanstalk.ElasticBeanstalk, d *schema.ResourceData) error {
 	name := d.Get("name").(string)
 	desc := d.Get("description").(string)
+	version := d.Get("version_label").(string)
```

Comment:
@zpatrick you're correct about the version update, 4c3fd07 fixes that. 

The answer to your second question is yes, terraform should update the environment with the new version as long as your environment resource references the new `version_label`. If you do something like the example document above, that will happen by changing the `version_label` for the `aws_elastic_beanstalk_application_version` resource. The output of `terraform plan` would look something like this:

```
-/+ aws_elastic_beanstalk_application_version.default
    application: "tf-test-name" => "tf-test-name"
    bucket:      "tftest.applicationversion.bucket" => "tftest.applicationversion.bucket"
    key:         "beanstalk/go-v1.zip" => "beanstalk/go-v1.zip"
    name:        "tf-test-version-label" => "tf-test-version-new-label" (forces new resource)

~ aws_elastic_beanstalk_environment.default
    version_label: "tf-test-version-label" => "tf-test-version-new-label"
```